Adding rsyslog topics back to 4.2

mburke5678 opened the pull request.

mburke5678 commented on modules/efk-logging-fluentd-json.adoc:
```diff
@@ -41,8 +45,8 @@ If you set the `MERGE_JSON_LOG` and `CDM_UNDEFINED_TO_STRING` enviroment variabl
 
```
@richm Does this problem affect Rsyslog as it affected Fluentd? 
`If you set the `MERGE_JSON_LOG` and `CDM_UNDEFINED_TO_STRING` enviroment variables to `true`, you might receive an Elasticsearch *400* error. `

richm replied:
Yes.

mburke5678 commented on modules/efk-logging-fluentd-throttling.adoc:
```diff
@@ -42,7 +42,7 @@ $ oc edit configmap/fluentd
++
 ----
 throttle-config.yaml: |
   - opensift-logging:
```
@richm Where in the config map is this code added? I couldn't get the code to be accepted.

richm replied:
Throttling is not supported (yet) by rsyslog.

mburke5678 commented on modules/efk-logging-fluentd-external.adoc:
```diff
@@ -13,6 +13,9 @@ hosted Fluentd has processed them.
+ifdef::openshift-enterprise[]
+The `secure-forward` plug-in is supported by Fluentd only.
+enddef::openshift-enterprise[]
 
```
@richm Is the `secure-forward` plug-in for Fluentd only?

richm replied:
Yes.  We'll have to figure out some other way for rsyslog to forward logs off of the cluster.

mburke5678 commented:
@richm This is a WIP. However, I have a few questions if you would be so kind.

richm commented on logging/efk-logging-eventrouter.adoc:
```diff
@@ -9,6 +9,11 @@ The Event Router communicates with the {product-title} and prints {product-title
 
+NOTE:
+====
+The Event Router is not supported for the Rsyslog log collector.
```
should add that to the list of differences between fluentd and rsyslog

mburke5678 replied:
Outdated

richm commented on modules/efk-logging-fluentd-log-rotation.adoc:
```diff
@@ -19,23 +19,34 @@ fluentd.log.3  1Mb
 
-| `LOGGING_FILE_SIZE` | The maximum size of a single Fluentd log file in Bytes. If the size of the *fluentd.log* file exceeds this value, {product-title} renames the *fluentd.log.** files and creates a new *fluentd.log*. The default is 1024000 (1MB).
-| `LOGGING_FILE_AGE` | The number of logs that Fluentd retains before deleting. The default value is `10`.
+| `LOGGING_FILE_SIZE` | The maximum size of a single log collector file in Bytes. If the size of the *fluentd.log* file exceeds this value, {product-title} renames the log files and creates a new file. The default is 1024000 (1MB).
```
size of the fluentd.log file or the rsyslog.log file

mburke5678 commented:
@QiaolingTang Is this something you can review? 
I reverted the current 4.1 docs to include both Rsyslog and Fluentd. 
* Changed _Fluentd_ to _log collector_ or _Fluentd or Rsyslog_ as appropriate.
* Added examples for Rsyslog as approptiate.
* Made a few other file names more generic.
* Other required changes.
The _might_ be some issues that are being followed in other PRs.

richm commented on modules/efk-logging-fluentd-envvar.adoc:
```diff
@@ -5,9 +5,13 @@
 configuration of the log collector, Fluentd or Rsyslog.
 
+See the link:https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/README.md[Fluentd README] in Github or the 
+link:https://github.com/openshift/origin-aggregated-logging/blob/master/rsyslog/README.md[Rsyslog README] for lists of the
```
https://github.com/openshift/origin-aggregated-logging/pull/1722

QiaolingTang commented on modules/efk-logging-about-components.adoc:
```diff
@@ -5,13 +5,13 @@
 
 * logStore - This is where the logs will be stored. The current implementation is Elasticsearch.
-* collection - This is the component that collects logs from the node, formats them, and stores them in the logStore. The current implementation is Fluentd.
+* collection - This is the component that collects logs from the node, formats them, and stores them in the logStore. The current implementation is Fluentd, default, or Rsyslog.
```
Seems we don't have an default option here.

mburke5678 replied:
@QiaolingTang Thank you. I will correct this throughout. There is no default. The user needs to specify the collector in the CR.

QiaolingTang commented on modules/efk-logging-configuring-image-about.adoc:
```diff
@@ -25,23 +25,11 @@ OAUTH_PROXY_IMAGE=registry.redhat.io/openshift4/ose-oauth-proxy:v4.1 <5>
-* *RSYSLOG_IMAGE* deploys Rsyslog, by default `docker.io/viaq/rsyslog:latest`. <1>
-
-<1> The image used for RSYSLOG when deployed. You can change this value using an environment variable. You cannot change this value through the Cluster Logging CR.
+<6> *RSYSLOG_IMAGE* deploys Rsyslog, by default `docker.io/viaq/rsyslog:latest`. You can change this value using an environment variable. You cannot change this value through the Cluster Logging CR.
```
I don't think the default value of RSYSLOG_IMAGE will be `docker.io/viaq/rsyslog:latest` after the rsyslog's GA. And the RSYSLOG_IMAGE is written in the CSV, it's better not to change the value. Setting environment variable won't make any change.

QiaolingTang commented on modules/efk-logging-fluentd-limits.adoc:
```diff
@@ -39,3 +39,27 @@ spec:
+          requests:
+            cpu: 250m
+            memory: 1Gi
+----
```
@mburke5678  I checked the default values of rsyslog's resources, they are:
    name: rsyslog
    resources:
      limits:
        memory: 358Mi
      requests:
        cpu: 100m
        memory: 358Mi

    name: logrotate
    resources:
      limits:
        memory: 358Mi
      requests:
        cpu: 100m
        memory: 358Mi

QiaolingTang commented on modules/efk-logging-fluentd-limits.adoc:
```diff
@@ -39,3 +39,27 @@ spec:
             memory: 1Gi
 ----
 <1> Specify the CPU and memory limits as needed. The values shown are the default values.
```
Users can specify the limits and requests.

QiaolingTang commented on modules/efk-logging-fluentd-log-location.adoc:
```diff
@@ -3,9 +3,9 @@
+= Configuring the collected log location
 
-Fluentd writes logs to a specified file or to the default location, `/var/log/fluentd/fluentd.log`, 
+The log collector, Fluentd or Rsyslog, writes logs to a specified file or to the default location, `/var/log/fluentd/fluentd.log`, 
```
For fluentd, it's /var/log/fluentd/fluentd.log , for rsyslog, it is /var/log/rsyslog/rsyslog.log

QiaolingTang commented on modules/efk-logging-fluentd-log-rotation.adoc:
```diff
@@ -19,23 +19,34 @@ fluentd.log.3  1Mb
+rsyslog.log.2  1Mb
+rsyslog.log.3  1Mb
+rsyslog.log.4  1Mb
+----
```
The log files for rsyslog is looks like:
-rw-r--r--. 1 root root      467 Jul 17 03:40 logrotate.log
-rw-r--r--. 1 root root     1007 Jul 17 03:25 rsyslog.log
-rw-r--r--. 1 root root 10382396 Jul 17 02:49 rsyslog.log-20190717-1563331501
-rw-r--r--. 1 root root 12649621 Jul 17 03:02 rsyslog.log-20190717-1563332101
-rw-r--r--. 1 root root 11345789 Jul 17 03:11 rsyslog.log-20190717-1563332701
-rw-r--r--. 1 root root 12271294 Jul 17 03:21 rsyslog.log-20190717-1563333302
-rw-r--r--. 1 root root 23122678 Jul 17 03:40 rsyslog.log-20190717-1563333901

mburke5678 replied:
Thank you. I forgot that I wanted an overly-simple example without the full naming to show the rotation. I'm not sure the fluentd log file names I have are accurate for what the user will see. 
Do you think this will work to show the rotation, or would using real log file names make more sense? 
```
The following example shows logs in a cluster where the maximum log size is 1Mb and four logs should be retained. When the `log`reaches 1Mb, {product-title}
deletes the current `log.4`, renames the each of the Fluentd logs in turn, and creates a new `log`.

----
log     0b
log.1  1Mb
log.2  1Mb
log.3  1Mb
log.4  1Mb
----
```

QiaolingTang commented on modules/efk-logging-fluentd-log-rotation.adoc:
```diff
@@ -19,23 +19,34 @@ fluentd.log.3  1Mb
 environment variables.
 
-.Parameters for configuring Fluentd log rotation
+.Parameters for configuring log rotation
```
The log rotation of rsyslog is different from fluentd. There has a configmap logrotate-crontab controlls the rsyslog's log rotation.  You can get some information from: https://bugzilla.redhat.com/show_bug.cgi?id=1730211#c1

QiaolingTang commented on modules/efk-logging-fluentd-log-rotation.adoc:
```diff
@@ -19,23 +19,34 @@ fluentd.log.3  1Mb
 
-| `LOGGING_FILE_SIZE` | The maximum size of a single Fluentd log file in Bytes. If the size of the *fluentd.log* file exceeds this value, {product-title} renames the *fluentd.log.** files and creates a new *fluentd.log*. The default is 1024000 (1MB).
-| `LOGGING_FILE_AGE` | The number of logs that Fluentd retains before deleting. The default value is `10`.
+| `LOGGING_FILE_SIZE` | The maximum size of of the fluentd.log file or the rsyslog.log file in Bytes. If the size of the *fluentd.log* file exceeds this value, {product-title} renames the log files and creates a new file. The default is 1024000 (1MB).
```
The log rotation of rsyslog is different from fluentd. There has a configmap logrotate-crontab controlls the rsyslog's log rotation. You can get some information from: https://bugzilla.redhat.com/show_bug.cgi?id=1730211#c1

richm replied:
Yes, there is a configmap with a crontab, and one with the logrotate config, but the user doesn't need to use those.  Most users will be able to set these environment variables and not need to know the implementation details.  I suppose we should document those configmaps, but we should tell users that they probably don't need to touch them.

mburke5678 replied:
@QiaolingTang @nhosoi In order to change the log rotation parameters, does the user still edit the `rsyslog` daemonset or edit the `logrotate` configmap?
```
oc set env daemonset/rsyslog LOGGING_FILE_AGE=30 LOGGING_FILE_SIZE=1024000

oc set env cm/logrotate LOGGING_FILE_AGE=30 LOGGING_FILE_SIZE=1024000
```

richm replied:
This one:
```
oc set env daemonset/rsyslog LOGGING_FILE_AGE=30 LOGGING_FILE_SIZE=1024000
```
If an advanced user needs to change the cron or logrotate settings, they would do
```
oc edit configmap logrotate-bin
or
oc edit configmap logrotate-crontab
```

mburke5678 replied:
@richm Thanks. I didn't see your comment above as I didn't refresh the page.

mburke5678 commented:
@QiaolingTang Are you satisfied with this PR?

QiaolingTang commented on modules/efk-logging-configuring-image-about.adoc:
```diff
@@ -14,34 +14,23 @@ You can view the images by running the following command:
+KIBANA_IMAGE=registry.redhat.io/openshift4/ose-logging-kibana5:v4.2 <3>
+CURATOR_IMAGE=registry.redhat.io/openshift4/ose-logging-curator5:v4.2 <4>
+OAUTH_PROXY_IMAGE=registry.redhat.io/openshift4/ose-oauth-proxy:v4.2 <5>
+RSYSLOG_IMAGE=quay.io/openshift/origin-logging-rsyslog:latest <6>
```
I thought the rsyslog image would be something like `registry.redhat.io/openshift4/ose-logging-rsyslog:v4.2` when the 4.2 GA.

QiaolingTang commented on modules/efk-logging-fluentd-alerts.adoc:
```diff
@@ -3,15 +3,32 @@
+= Viewing collected logs
 
-How you view logs depends upon the `LOGGING_FILE_PATH` setting.
+How you view logs collected by the log collector, Fluentd or Rsyslog, depends upon the `LOGGING_FILE_PATH` setting.
```
These logs are not collected by the log collector, they are generated by the log collector.

mburke5678 commented:
Moving to 4.3. Reverting changes added to 4.2 

https://github.com/openshift/openshift-docs/pull/16597
